source/process/regex-library-draft.md

```diff
@@ -0,0 +1,175 @@
+
+</details>
+
+<details close>
+
+<summary>Links</summary>
+
+* **add `target="_blank"` to links**: Add `target="_blank"` attribute to existing external links<br>F: `<a href="http([^"]+)"><br>R: <a href="http\1" target="_blank" rel="noopener">`<br>R: `<a href="http\1" target="_blank" rel="noopener">`
+* **URLs**: Add links to URLs (Does not capture every instance)<br>F: `\shttp(.+?)([;|\.|,|\)][\s|<])`<br>R: `\s<a href="http\1" target="_blank" rel="noopener">http\1</a>\2\3`
+* **tag hyperlinks:** find and replace to tag hyperlinks<br>F: `<a (?:class="[^"]*"\s*)*href="((?:mail[^"]*)|(?:http[^"]*))">([^<]*)</a>`<br>R: `<a href="\1" target="_blank" rel="noopener">\2</a>`
+* **link chapters**: Find potential instances where chapters can be linked. Adjust the word `first` to `second` and the number `1` to `2` etc., to find all chapters<br>F: `(first chap(\.|ters?)|chap(s?\.|ters?) 1)(?!\d)`
+* **link parts**: Find potential instances where parts can be linked. Adjust the word `first` to `second` and the number `1` to `2` etc., to find all parts<br>F: `(first part|parts? 1)(?!\d)`
+
+</details>
+
+<details close>
+
+<summary>Percival</summary>
+
+* **percival parsing**: add parsing tags before headings containing scripture. Replace `Gen` with Bible book needed<br>F: `^(\s+)<(h\d)>(.*?)(\d+):(.*?)</\2>`<br>R: `\1<span data-parsing="Gen.\4"></span>\n\1<\2>\3\4:\5</\2>`
+
+</details>
+
+<details close>
+
+<summary>Commentary Markup</summary>
+
+* **headings `data-context`**: add `data-context` tags before headings. Adjust `h3` to capture desired heading<br>F: `^(\s+)<(h3)>(.*?<a data-ref="(.*?)">.*?</a>.*?)</\2>`<br>R: `\1<hr data-context="\4" />\n\1<\2>\3</\2>`
+
+</details>
+
+<details close>
+
+<summary>LESSON\_maker\_ Markup</summary>
+
+* **panic LESSON_maker_**: add the `panic` attribute to list items with `data-refs`<br>F: `<li( value="\d+")? data-question(="\d+">.*?<a data-ref=")`<br>R: `<li\1 data-panic\2`
+
+</details></blockquote>
+
+</details>
+
+<br>
+
+<details close>
+
+<summary>Review</summary>
+
+* **remove pagebreaks from headings: **find and replace to move pagebreaks out of headings<br>F: `(<h\d>.*?)(<span epub:type="pagebreak[^>]*></span>)`<br>R: `\2\1`
+* **remove space before footnote**: find and replace extra space before a footnote indicator<br>F: `\s<sup class="fn"`<br>R: `<sup class="fn"`
+* **special chars spacing: **review special character spacing<br>F: `\s+(\{|\$|\&|\,|\:|\;|\?|\@|\#|\||\'|\<|\>|\-|\^|\*|\(|\)|\%|\!|\]|\"|”|“)\s+`<br>R: `\2 \1`
+* **special chars spans: **review special characters in spans and replace the character without the span<br>F: `<span[^>]>({|$|&|,|:|;|?|@|#|||'|.|-|^||(|)|%|!|]|"|”|“|—)+</span>`<br>R: `\1`
+* **non-english chars spans: **review non-english characters in spans that could be tagged as `lang`<br>F: `<span class="i">([^a-zA-Z0-9\s]+)</span>`
```

Comment:
maybe add `italic` as an alternative class in order to expand what this RegEx will match?
```
<span class="i(?:talic)?">([^a-zA-Z0-9\s]+)</span>
```